File: sdks/sdkwork-craw-chat-sdk-admin/sdkwork-craw-chat-sdk-admin-typescript/composed/src/social-module.ts
import type { Identifier, JsonObject } from './types.js';
import type { CrawChatAdminSdkContext } from './sdk-context.js';

export class CrawChatAdminSocialModule {
  constructor(private readonly context: CrawChatAdminSdkContext) {}

  bindDirectChat(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.social.bindDirectChat(body);
  }

  getDirectChat(id: Identifier): Promise<JsonObject> {
    return this.context.backendClient.social.getDirectChatSnapshot(id);
  }

  establishExternalConnection(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.social.establishExternalConnection(body);
  }

  getExternalConnection(id: Identifier): Promise<JsonObject> {
    return this.context.backendClient.social.getExternalConnectionSnapshot(id);
  }

  bindExternalMemberLink(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.social.bindExternalMemberLink(body);
  }

  getExternalMemberLink(id: Identifier): Promise<JsonObject> {
    return this.context.backendClient.social.getExternalMemberLinkSnapshot(id);
  }

  submitFriendRequest(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.social.submitFriendRequest(body);
  }

  getFriendRequest(id: Identifier): Promise<JsonObject> {
    return this.context.backendClient.social.getFriendRequestSnapshot(id);
  }

  activateFriendship(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.social.activateFriendship(body);
  }

  getFriendship(id: Identifier): Promise<JsonObject> {
    return this.context.backendClient.social.getFriendshipSnapshot(id);
  }

  applySharedChannelPolicy(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.social.applySharedChannelPolicy(body);
  }

  getSharedChannelPolicy(id: Identifier): Promise<JsonObject> {
    return this.context.backendClient.social.getSharedChannelPolicySnapshot(id);
  }

  blockUser(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.social.blockUser(body);
  }

  getUserBlock(id: Identifier): Promise<JsonObject> {
    return this.context.backendClient.social.getUserBlockSnapshot(id);
  }
}

File: sdks/sdkwork-control-plane-sdk/sdkwork-control-plane-sdk-flutter/composed/lib/src/social_module.dart
import 'package:control_plane_backend_sdk/control_plane_backend_sdk.dart';

import 'context.dart';

class ControlPlaneSocialModule {
  final ControlPlaneSdkContext context;

  ControlPlaneSocialModule(this.context);

  Future<JsonObject> bindDirectChat(JsonObject body) {
    return context.backendClient.social.bindDirectChat(body);
  }

  Future<JsonObject> getDirectChat(Identifier id) {
    return context.backendClient.social.getDirectChatSnapshot(id);
  }

  Future<JsonObject> establishExternalConnection(JsonObject body) {
    return context.backendClient.social.establishExternalConnection(body);
  }

  Future<JsonObject> getExternalConnection(Identifier id) {
    return context.backendClient.social.getExternalConnectionSnapshot(id);
  }

  Future<JsonObject> bindExternalMemberLink(JsonObject body) {
    return context.backendClient.social.bindExternalMemberLink(body);
  }

  Future<JsonObject> getExternalMemberLink(Identifier id) {
    return context.backendClient.social.getExternalMemberLinkSnapshot(id);
  }

  Future<JsonObject> submitFriendRequest(JsonObject body) {
    return context.backendClient.social.submitFriendRequest(body);
  }

  Future<JsonObject> getFriendRequest(Identifier id) {
    return context.backendClient.social.getFriendRequestSnapshot(id);
  }

  Future<JsonObject> activateFriendship(JsonObject body) {
    return context.backendClient.social.activateFriendship(body);
  }

  Future<JsonObject> getFriendship(Identifier id) {
    return context.backendClient.social.getFriendshipSnapshot(id);
  }

  Future<JsonObject> applySharedChannelPolicy(JsonObject body) {
    return context.backendClient.social.applySharedChannelPolicy(body);
  }

  Future<JsonObject> getSharedChannelPolicy(Identifier id) {
    return context.backendClient.social.getSharedChannelPolicySnapshot(id);
  }

  Future<JsonObject> blockUser(JsonObject body) {
    return context.backendClient.social.blockUser(body);
  }

  Future<JsonObject> getUserBlock(Identifier id) {
    return context.backendClient.social.getUserBlockSnapshot(id);
  }
}
